src/Kiota.Builder/CodeDOM/CodeClass.cs

```diff
@@ -65,12 +67,122 @@ public bool IsErrorDefinition
+            if (element is CodeProperty propertyToAdd)
+            {
+                PropertiesByWireName.TryAdd(propertyToAdd.WireName, propertyToAdd);
+            }
+        }
+    }
+    public override void RemoveChildElementByName(params string[] names)
+    {
+        if (names == null) return;
+
+        foreach (var name in names)
+        {
+            if (InnerChildElements.TryRemove(name, out var removedElement) && removedElement is CodeProperty removedProperty)
+            {
+                PropertiesByWireName.TryRemove(removedProperty.WireName, out _);
+            }
+        }
+    }
+    private string ResolveUniquePropertyName(string name)
+    {
+        if (FindPropertyByNameInTypeHierarchy(name) == null)
+            return name;
+        // the CodeClass.Name is not very useful as prefix for the property name, so keep the original name and add a number
+        var nameWithTypeName = Kind == CodeClassKind.QueryParameters ? name : Name + name.ToFirstCharacterUpperCase();
+        if (Kind != CodeClassKind.QueryParameters)
+        {
+            if (FindPropertyByNameInTypeHierarchy(nameWithTypeName) == null)
+                return nameWithTypeName;
+        }
+        var i = 0;
+        while (FindPropertyByNameInTypeHierarchy(nameWithTypeName + i) != null)
+            i++;
+        return nameWithTypeName + i;
+    }
+    private CodeProperty? FindPropertyByNameInTypeHierarchy(string propertyName)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(propertyName);
+
+        if (FindChildByName<CodeProperty>(propertyName, findInChildElements: false) is CodeProperty result)
+        {
+            return result;
+        }
+        if (ParentClass is CodeClass currentParentClass)
+        {
+            return currentParentClass.FindPropertyByNameInTypeHierarchy(propertyName);
+        }
+        return default;
+    }
+    private CodeProperty? GetOriginalPropertyDefinedFromBaseType(string serializationName)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(serializationName);
+
+        if (ParentClass is CodeClass currentParentClass)
+            if (currentParentClass.FindPropertyByWireName<CodeProperty>(serializationName) is CodeProperty currentProperty && !currentProperty.ExistsInBaseType)
+                return currentProperty;
+            else
+                return currentParentClass.GetOriginalPropertyDefinedFromBaseType(serializationName);
+        return default;
+    }
+    private CodeProperty? FindPropertyByWireName<T>(string wireName)
+    {
+        if (!PropertiesByWireName.Any())
+            return default;
+
+        if (PropertiesByWireName.TryGetValue(wireName, out var result))
+            return result;
+        foreach (var childElement in InnerChildElements.Values.OfType<CodeClass>())
+        {
+            var childResult = childElement.FindPropertyByWireName<T>(wireName);
+            if (childResult != null)
+                return childResult;
+        }
+        return default;
```

Comment:
```suggestion
        return InnerChildElements.Values.OfType<CodeClass>().FirstOrDefault(x => x.FindPropertyByWireName<T>(wireName));
```

Reply:
Your suggestion didn't compile because the result of `x.FindPropertyByWireName` is not a boolean, I changed it to this: ```return InnerChildElements.Values.OfType<CodeClass>().Select(x => x.FindPropertyByWireName(wireName)).Where(x => x != null).FirstOrDefault();```